app/AppKernel.php

```diff
@@ -148,68 +137,20 @@ public function registerContainerConfiguration(LoaderInterface $loader)
      * Return all active modules.
      *
      * @return array list of modules names.
-     * @throws \Doctrine\DBAL\DBALException
      */
     private function getActiveModules()
     {
-        $databasePrefix = $this->getParameters()['database_prefix'];
-
-        $modulesRepository = new ModuleRepository(
-            $this->getConnection(),
-            $databasePrefix
-        );
-
-        return $modulesRepository->getActiveModules();
-    }
-
-    /**
-     * @return array The root parameters of PrestaShop.
-     */
-    private function getParameters()
-    {
-        if ($this->parametersFileExists()) {
-            $config = require $this->getParametersFile();
-
-            return $config['parameters'];
+        $activeModules = [];
+        try {
+            if ($modulesRepository = ModuleRepositoryFactory::getInstance()->getRepository()) {
+                $activeModules = $modulesRepository->getActiveModules();
+            }
+        } catch (\Exception $e) {
+            //Do nothing because the modules retrieval must not block the kernel
```

Comment:
Why not? Why do you want to catch this exception?

Reply:
Agreed, if you add a module which break everything, you must know why.

Reply:
I agree with you, I just based this code on the original one where exceptions were caught, maybe to avoid errors during the installation process when module are not available yet? or in case the database connection is not correctly set
https://github.com/PrestaShop/PrestaShop/blob/75dda346dd95d489c7ae56aeb1d9ab3f33ac841b/app/AppKernel.php#L98

As we are in the kernel if we throw an exception here it can block a lot of processes

Reply:
oh f***, I remember yeah: it was done to manage installation step ... can you put a comment?

Reply:
I did but seems like the comment was not clear enough ^^

Reply:
I'll update it